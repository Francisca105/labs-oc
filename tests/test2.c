#include "../code/4.1/L1Cache.h"

#define offset_bits 6
#define index_bits 8

// Checks if is there any problem with alignment and if a block is placed on cache on full extent

// The output might seem strange but those leaps between stored data happen because when a write -
// or read for that matter - happens, it is written a full aligned word on the memory, meaning
// only the last value will stay stored
// Despite the memory being theoretically byte addressable, ensuring the data alignment makes it
// so that in reality it is only word addressable

int main() {

  resetTime();
  initCache();

  int i, value;

  for (i = 0; i < BLOCK_SIZE; i++) {
    write(i, (unsigned char *)(&i));
    printf("Write; Address %d; Value %d; Time %d\n", i, i, getTime());
  }

  write(i, (unsigned char *)(&i));
  printf("\nWrite 1 byte after the end of the block\n");
  printf("Write; Address %d; Value %d; Time %d\n", i, i, getTime());
  printf("Confirmed that block ended due to necessity of going to the main memory\n");

  i = 1 << (offset_bits + index_bits);
  write(i, (unsigned char *)(&i));
  printf("\nWrite; Address %d; Value %d; Time %d\n", i, i, getTime());
  printf("Replaced the block\n\n");

  for (i = 0; i < BLOCK_SIZE; i++) {
    read(i, (unsigned char *)(&value));
    printf("Read; Address %d; Value %d; Time %d\n", i, value, getTime());
  }
  
  return 0;
}