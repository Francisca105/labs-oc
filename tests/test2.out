Write; Address 0; Value 0; Time 101
Write; Address 1; Value 1; Time 102
Write; Address 2; Value 2; Time 103
Write; Address 3; Value 3; Time 104
Write; Address 4; Value 4; Time 105
Write; Address 5; Value 5; Time 106
Write; Address 6; Value 6; Time 107
Write; Address 7; Value 7; Time 108
Write; Address 8; Value 8; Time 109
Write; Address 9; Value 9; Time 110
Write; Address 10; Value 10; Time 111
Write; Address 11; Value 11; Time 112
Write; Address 12; Value 12; Time 113
Write; Address 13; Value 13; Time 114
Write; Address 14; Value 14; Time 115
Write; Address 15; Value 15; Time 116
Write; Address 16; Value 16; Time 117
Write; Address 17; Value 17; Time 118
Write; Address 18; Value 18; Time 119
Write; Address 19; Value 19; Time 120
Write; Address 20; Value 20; Time 121
Write; Address 21; Value 21; Time 122
Write; Address 22; Value 22; Time 123
Write; Address 23; Value 23; Time 124
Write; Address 24; Value 24; Time 125
Write; Address 25; Value 25; Time 126
Write; Address 26; Value 26; Time 127
Write; Address 27; Value 27; Time 128
Write; Address 28; Value 28; Time 129
Write; Address 29; Value 29; Time 130
Write; Address 30; Value 30; Time 131
Write; Address 31; Value 31; Time 132
Write; Address 32; Value 32; Time 133
Write; Address 33; Value 33; Time 134
Write; Address 34; Value 34; Time 135
Write; Address 35; Value 35; Time 136
Write; Address 36; Value 36; Time 137
Write; Address 37; Value 37; Time 138
Write; Address 38; Value 38; Time 139
Write; Address 39; Value 39; Time 140
Write; Address 40; Value 40; Time 141
Write; Address 41; Value 41; Time 142
Write; Address 42; Value 42; Time 143
Write; Address 43; Value 43; Time 144
Write; Address 44; Value 44; Time 145
Write; Address 45; Value 45; Time 146
Write; Address 46; Value 46; Time 147
Write; Address 47; Value 47; Time 148
Write; Address 48; Value 48; Time 149
Write; Address 49; Value 49; Time 150
Write; Address 50; Value 50; Time 151
Write; Address 51; Value 51; Time 152
Write; Address 52; Value 52; Time 153
Write; Address 53; Value 53; Time 154
Write; Address 54; Value 54; Time 155
Write; Address 55; Value 55; Time 156
Write; Address 56; Value 56; Time 157
Write; Address 57; Value 57; Time 158
Write; Address 58; Value 58; Time 159
Write; Address 59; Value 59; Time 160
Write; Address 60; Value 60; Time 161
Write; Address 61; Value 61; Time 162
Write; Address 62; Value 62; Time 163
Write; Address 63; Value 63; Time 164

Write 1 byte after the end of the block
Write; Address 64; Value 64; Time 265
Confirmed that block ended due to necessity of going to the main memory

Write; Address 16384; Value 16384; Time 416
Replaced the block

Read; Address 0; Value 3; Time 567
Read; Address 1; Value 3; Time 568
Read; Address 2; Value 3; Time 569
Read; Address 3; Value 3; Time 570
Read; Address 4; Value 7; Time 571
Read; Address 5; Value 7; Time 572
Read; Address 6; Value 7; Time 573
Read; Address 7; Value 7; Time 574
Read; Address 8; Value 11; Time 575
Read; Address 9; Value 11; Time 576
Read; Address 10; Value 11; Time 577
Read; Address 11; Value 11; Time 578
Read; Address 12; Value 15; Time 579
Read; Address 13; Value 15; Time 580
Read; Address 14; Value 15; Time 581
Read; Address 15; Value 15; Time 582
Read; Address 16; Value 19; Time 583
Read; Address 17; Value 19; Time 584
Read; Address 18; Value 19; Time 585
Read; Address 19; Value 19; Time 586
Read; Address 20; Value 23; Time 587
Read; Address 21; Value 23; Time 588
Read; Address 22; Value 23; Time 589
Read; Address 23; Value 23; Time 590
Read; Address 24; Value 27; Time 591
Read; Address 25; Value 27; Time 592
Read; Address 26; Value 27; Time 593
Read; Address 27; Value 27; Time 594
Read; Address 28; Value 31; Time 595
Read; Address 29; Value 31; Time 596
Read; Address 30; Value 31; Time 597
Read; Address 31; Value 31; Time 598
Read; Address 32; Value 35; Time 599
Read; Address 33; Value 35; Time 600
Read; Address 34; Value 35; Time 601
Read; Address 35; Value 35; Time 602
Read; Address 36; Value 39; Time 603
Read; Address 37; Value 39; Time 604
Read; Address 38; Value 39; Time 605
Read; Address 39; Value 39; Time 606
Read; Address 40; Value 43; Time 607
Read; Address 41; Value 43; Time 608
Read; Address 42; Value 43; Time 609
Read; Address 43; Value 43; Time 610
Read; Address 44; Value 47; Time 611
Read; Address 45; Value 47; Time 612
Read; Address 46; Value 47; Time 613
Read; Address 47; Value 47; Time 614
Read; Address 48; Value 51; Time 615
Read; Address 49; Value 51; Time 616
Read; Address 50; Value 51; Time 617
Read; Address 51; Value 51; Time 618
Read; Address 52; Value 55; Time 619
Read; Address 53; Value 55; Time 620
Read; Address 54; Value 55; Time 621
Read; Address 55; Value 55; Time 622
Read; Address 56; Value 59; Time 623
Read; Address 57; Value 59; Time 624
Read; Address 58; Value 59; Time 625
Read; Address 59; Value 59; Time 626
Read; Address 60; Value 63; Time 627
Read; Address 61; Value 63; Time 628
Read; Address 62; Value 63; Time 629
Read; Address 63; Value 63; Time 630
